src/net_processing.cpp

```diff
@@ -4343,7 +4422,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         } // cs_main
 
         if (fProcessBLOCKTXN) {
-            return ProcessMessage(pfrom, NetMsgType::BLOCKTXN, blockTxnMsg, time_received, interruptMsgProc);
+            return ProcessBlockTxnMessage(pfrom, blockTxnMsg, *peer, msgMaker);
```

Comment:
This isn't really processing a BLOCKTXN message though -- it's completing the block immediately because a BLOCKTXN message wasn't necessary?

Wouldn't it be better to call it `ProcessCompactBlockTxns` and pass in the `BlockTransactions` directly, rather than serializing then unserializing it?

Reply:
> Wouldn't it be better to call it ProcessCompactBlockTxns and pass in the BlockTransactions directly, rather than serializing then unserializing it?

I think you're right, it would be better to pass `BlockTransactions` directly and not deal with serialization.